stock_available_to_promise_release_carrier_alternative/models/stock_picking.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _apply_alternative_carrier(self):
+        carrier = self._get_preferred_carrier()
+        if not carrier:
+            return False
+        if carrier == self.carrier_id:
+            return False
+        # Set a context key to not trigger a carrier change on the
+        # procurement group
+        self.with_context(skip_align_group_carrier=True).carrier_id = carrier
+        self.group_id = self.group_id.copy(default={"carrier_id": carrier.id})
```

Comment:
The moves must have the new group otherwise the pull flow won't use that new group and carrier
```suggestion
        self.group_id = self.group_id.copy(default={"carrier_id": carrier.id})
        self.move_ids.filtered(lambda m: m.state not in ("done", "cancel")).group_id = self.group_id
```